File: app/src/main/java/com/fthdgn/books/dependecy/component/AppComponent.java
package com.fthdgn.books.dependecy.component;

import android.view.inputmethod.InputMethodManager;

import javax.inject.Singleton;

import dagger.Component;
import com.fthdgn.books.api.AccountManager;
import com.fthdgn.books.api.GoogleBooksApi;
import com.fthdgn.books.api.ImageManager;
import com.fthdgn.books.dependecy.module.ApiModule;
import com.fthdgn.books.dependecy.module.AppModule;
import com.fthdgn.books.dependecy.module.NetworkModule;
import com.fthdgn.books.dependecy.module.PicassoModule;
import com.fthdgn.books.dependecy.module.PreferencesModule;
import com.fthdgn.books.dependecy.module.RepositoryModule;
import com.fthdgn.books.repository.BookDao;

@Singleton
@Component(modules = {
        ApiModule.class,
        AppModule.class,
        NetworkModule.class,
        PreferencesModule.class,
        RepositoryModule.class,
        PicassoModule.class})
public interface AppComponent {

    BookDao bookDao();

    GoogleBooksApi googleBooksApi();

    AccountManager accountManager();

    InputMethodManager inputMethodManager();

    ImageManager imageManager();

}

File: app/src/main/java/com/fthdgn/books/dependecy/component/AppComponent.kt
package com.fthdgn.books.dependecy.component

import com.fthdgn.books.api.AccountManager
import com.fthdgn.books.api.GoogleBooksApi
import com.fthdgn.books.api.ImageManager
import com.fthdgn.books.dependecy.module.*
import com.fthdgn.books.repository.BookDao
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApiModule::class, AppModule::class, NetworkModule::class, PreferencesModule::class, RepositoryModule::class, PicassoModule::class))
interface AppComponent {

    fun bookDao(): BookDao

    fun googleBooksApi(): GoogleBooksApi

    fun accountManager(): AccountManager

    fun imageManager(): ImageManager

}
